midori/sokoke.c is using XGetWindowProperty() and XFree(), therefore link against -lX11 to prevent
underlinking failures like "error: undefined reference to 'XGetWindowProperty'" and
"error: undefined reference to 'XFree'"

$ ld -v 
GNU gold (GNU Binutils 2.24) 1.11
    ^^^^

$ emerge --info |grep LDFLAGS
LDFLAGS="-Wl,-O1 -Wl,--as-needed -Wl,--hash-style=gnu"
                 ^^^^^^^^^^^^^^^

--- midori-0.5.7.orig/CMakeLists.txt	2014-01-15 22:43:34.000000000 +0200
+++ midori-0.5.7/CMakeLists.txt	2014-03-29 13:06:16.714114071 +0200
@@ -77,7 +77,7 @@
     find_library (XSS Xss /usr/lib/X11R6/lib)
     if (SCRNSAVER AND XSS)
         add_definitions("-DHAVE_X11_EXTENSIONS_SCRNSAVER_H")
-        set(OPTS_LIBRARIES "${OPTS_LIBRARIES};${XSS}")
+        set(OPTS_LIBRARIES "${OPTS_LIBRARIES};${XSS};${X11}")
     endif ()
 endif ()
 
